pkg/plugins/policies/meshloadbalancingstrategy/api/v1alpha1/validator.go

```diff
@@ -82,17 +83,32 @@ func validateTo(topTargetRef common_api.TargetRef, to []To) validators.Validatio
 
 func validateConf(conf Conf, to To) validators.ValidationError {
 	var verr validators.ValidationError
-	verr.AddError("loadBalancer", validateLoadBalancer(conf.LoadBalancer))
-	verr.AddError("localityAwareness", validateLocalityAwareness(conf.LocalityAwareness, to))
-	verr.AddError("hashPolicies", validateHashPolicies(conf.HashPolicies))
 
-	// Check if hashPolicies is specified both at the top level and in one of the load balancer types
-	if conf.HashPolicies != nil && conf.LoadBalancer != nil {
-		if conf.LoadBalancer.RingHash != nil && conf.LoadBalancer.RingHash.HashPolicies != nil {
-			verr.AddViolation("loadBalancer.ringHash.hashPolicies", "hashPolicies already specified in the root level")
+	// For MeshHTTPRoute, only hashPolicies is allowed
+	if to.TargetRef.Kind == common_api.MeshHTTPRoute {
+		verr.AddError("hashPolicies", validateHashPolicies(conf.HashPolicies))
+
+		// Add validation errors if other fields are specified
+		if conf.LoadBalancer != nil {
+			verr.AddViolation("loadBalancer", "field is not allowed when targetRef.kind is MeshHTTPRoute, only hashPolicies is supported")
+		}
+		if conf.LocalityAwareness != nil {
```

Comment:
Question: Why don't we want to allow configuring LocalityAwerness? usually when we have a MeshHTTPRoute we create a separate Cluster and that enables us to configure weight and priority for each endpoint